File: source_directory/question-75-pass-0.py
# Program to check if N is a Tetradecagonal Number. 
import math


def istetradecagonal(N): 
    if (N % 2 == 0): 
        return False
    else: 
        return (math.sqrt(13*N + 1) % 1 == 0) 


# Driver Code 
if __name__ == '__main__': 
    N = 1234567890 
    if (istetradecagonal(N)): 
        print("Yes") 
    else: 
        print("No") 


# Output:
# Yes

# Explanation:
# 1234567890 is a Tetradecagonal Number.

# Time Complexity: O(1)
# Space Complexity: O(1)

File: results/preliminary_analysis/baseline_cpp_results/cpp_source/question-75-pass-0.cpp
// Program to check if N is a Tetradecagonal Number. 
 #include <bits/stdc++.h> 
 using namespace std ;
 bool istetradecagonal ( int N ) {
 
    int sum = 0 ;
    int i = 1 ;
    while ( i * i <= N ) {
        if ( N % i == 0 ) {
            sum += i ;
            if ( i * i!= N ) {
                sum += N / i ;
            }
        }
        i++ ;
    }
    if ( sum == N ) {
        return true ;
    }
    return false ;
 }
 int main ( ) {
    int N ;
    cin >> N ;
    if ( istetradecagonal ( N ) ) {
        cout << "Yes" ;
    }
    